CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# Contributing
+
+This file provides general guidance for developers that are creating or updating REST API definitions for Azure.
+
+## Table of Contents
+<!--
+  You should regenerate the TOC using the `markdown-toc` node package after making changes to this file.
+
+      npx markdown-toc -i --maxdepth 4 CONTRIBUTING.md
+  -->
+
+<!-- toc -->
+
+- [Reporting problems](#reporting-problems)
+- [Avoid Breaking Changes](#avoid-breaking-changes)
+- [Design Guidelines](#design-guidelines)
+  * [Exceptions for consistency within a service](#exceptions-for-consistency-within-a-service)
+- [Coding Style](#coding-style)
+- [Directory Structure](#directory-structure)
+- [Pull Requests](#pull-requests)
+- [Pull Request checks](#pull-request-checks)
+
+<!-- tocstop -->
+
+## Reporting problems
+
+If you discover a problem with a REST API document in this repo, feel free to [open an issue](https://github.com/Azure/azure-rest-api-specs/issues/new). But please do not report issues with service behavior / functionality in this repo.
+
+When opening an issue, please clearly describe the problem including which REST API definition is in error and what elements of the definition are incorrect or missing.
+
+## Avoid Breaking Changes
+
+The first rule for Azure REST APIs is to avoid breaking changes in Generally Available (GA'd) APIs. The [Azure Breaking Changes Policy](https://aka.ms/AzBreakingChangesPolicy) is a formal description of what changes are considered breaking changes. This rule trumps adherence to Design Guidelines, Coding Style, etc.
+
+## Design Guidelines
+
+Azure REST APIs should conform to the [Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md).
+
+There is a [YouTube video series](https://www.youtube.com/watch?v=9Ng00IlBCtw) by Jeffrey Richter that describes the guidelines in detail.
+
+Another resource is the [Considerations for Service Design](https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md), which is an introduction to REST API design mainly for services that are just getting started.
+
+### Exceptions for consistency within a service
+
+There are situations where a service has GA'd their API with design patterns that do not follow our guidelines and it would be a breaking change to adopt the API design in the guidelines.
+Because the first rule is to avoid breaking changes and because we want APIs to be consistent within a service, these design patterns are considered the standard for that service and should be followed in all subsequent (non-breaking) versions of that service's REST API.
+
+## Coding Style
+
+REST APIs for Azure should be defined using the [OpenAPI v2](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) format.
```

Comment:
wonder if we should recommend Cadl here for Cadl libraries contain rules that encourage following established patterns for Azure APIs.  The document can be referred is https://github.com/Azure/cadl-azure/blob/main/docs/cadl-getting-started.md